taskana-adapter/src/main/java/pro/taskana/adapter/configuration/OutboxSchemaCreator.java

```diff
@@ -0,0 +1,161 @@
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(this.getClass()
+                .getResourceAsStream(selectDbSchemaDetectionScript(productName))));
+            runner.runScript(getSqlSchemaNameParsed(reader, productName));
+        } catch (Exception e) {
+            LOGGER.debug("Schema does not exist.");
+            return false;
+        }
+        LOGGER.debug("Schema does exist.");
+        return true;
+    }
+
+    public boolean isValidSchemaVersion(String expectedVersion) {
+        SqlRunner runner = null;
+        try {
+            Connection connection = dataSource.getConnection();
+            // connection.setSchema(this.schemaName);
+
+            runner = new SqlRunner(connection);
+            LOGGER.debug(connection.getMetaData().toString());
+
+            String query = "select VERSION from taskana_tables.OUTBOX_SCHEMA_VERSION where "
+                + "VERSION = (select max(VERSION) from taskana_tables.OUTBOX_SCHEMA_VERSION) "
+                + "AND VERSION = ?";
+
+            Map<String, Object> queryResult = runner.selectOne(query, expectedVersion);
+            if (queryResult == null || queryResult.isEmpty()) {
+                LOGGER.error(
+                    "Schema version not valid. The VERSION property in table TASKANA_SCHEMA_VERSION has not the expected value {}",
+                    expectedVersion);
+                return false;
+            } else {
+                LOGGER.debug("Schema version is valid.");
+                return true;
+            }
+
+        } catch (Exception e) {
+            LOGGER.error(
+                "Schema version not valid. The VERSION property in table TASKANA_SCHEMA_VERSION has not the expected value {}",
+                expectedVersion);
+            return false;
+        } finally {
+            if (runner != null) {
+                runner.closeConnection();
+            }
+        }
+    }
+
+    public DataSource getDataSource() {
+        return dataSource;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    private StringReader getSqlSchemaNameParsed(BufferedReader reader, String dbProductName) {
+        boolean isPostGres = TaskanaEngineImpl.isPostgreSQL(dbProductName);
+        StringBuffer content = new StringBuffer();
+        String effectiveSchemaName = isPostGres ? schemaName.toLowerCase() : schemaName.toUpperCase();
+        try {
+            String line = "";
+            while (line != null) {
+                line = reader.readLine();
+                if (line != null) {
+                    content.append(line.replaceAll("%schemaName%", effectiveSchemaName) + System.lineSeparator());
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error("SchemaName sql parsing failed for schemaName {}. Caught {}", effectiveSchemaName, e);
+        }
+        return new StringReader(content.toString());
+    }
+}
```

Comment:
A lot of copy / paste from AdapterSchemaCreator -> if schema creation is really performed through the adapter instead of e.g. the camunda ProcessEngine, then we should maybe think of a common base class from which common code can be inherited (DRY)

Reply:
This class is just a quick hack to enable taskana-adapter-test to run on travis. The final solution will definitely look different. By the way: this class does not belong to project taskana-adapter. I moved it to project taskana-adapter-camunda-outbox-rest-spring-boot...